technical-guidelines.md

```diff
@@ -0,0 +1,120 @@
+| semantic versioning, changelogs                | [Semantic Versioning][semantic-versioning], [Conventional Commits][conventional-commits], [Release Please][release-please]\*\*\*          | required                                                    |
+| unit, integration testing                      | github actions (with permissions applying principle of least privilege), language-specific tools (JUnit, Jest, etc), [Cucumber][cucumber] | required, with coverage metrics up to maintainer discretion |
+| signing (binaries, packages, container images) | language-specific tools, [cosign][cosign]                                                                                                 | recommended, where supported                                |
+| fuzzing                                        | [ClusterFuzzLite][clusterfuzzlite], [OSS-Fuzz][oss-fuzz]                                                                                  | recommended                                                 |
+| helpful readme file                            | See [example README.md](./templates/READMEs/README.md)                                                                                    | recommended                                                 |
+| provenance                                     | [SLSA](https://slsa.dev/spec/v1.0/provenance#provenance)                                                                                  | recommended                                                 |
+| SBOM generation                                | [CycloneDX][cyclonedx], [SPDX][spdx], [syft][syft]                                                                                        | recommended                                                 |
+| static analysis                                | [SonarCloud][sonarcloud], language-specific tools (SpotBugs, eslint)                                                                      | recommended                                                 |
+
+## Semantic Versioning and 1.0 Releases
+
+We require release artifacts to adhere to [semantic versioning](https://semver.org/) except in specific cases approved by the [Technical Steering Committee](https://github.com/open-feature/community/blob/main/community-members.md#technical-committee).
+1.0 releases must satisfy the [repository requirements](#repository-requirements) above.
+If the artifact is an SDK implementing the OpenFeature specification, it must also conform to a version of the OpenFeature specification not less than 2 minor versions behind the latest (ex: if the latest OpenFeature specification is `0.8.1`, then an implementation conforming to `0.6.0` is a candidate for `1.0` release, while an implementation conforming only to `0.5.0` is not).
```

Comment:
This was a bit of a point of contention in the TC meeting, but I think a "moving target" is better than a static one (`0.6.0`). Open to further discussion here, of course.

Reply:
There's a lot of importance being placed on 1.0 specifically, so it seems worth noting that while OF spec is 0.x, minor releases can have backwards incompatible changes. Releasing 1.0 may necessitate 2.0 for sdk authors.

Reply:
Wouldn't it be reasonable for a new SDK to be fully spec complient? We don't update the spec that often.

Reply:
Do we want to make any differentiation between sections of the spec? Is an experimental section still required for a 1.0 release? Or only once it's promoted to hardening?

Reply:
This is a good point. I've tried to address this with additional changes:

```markdown
> [!IMPORTANT]  
> While the OpenFeature specification version is <`1.0`, it's possible breaking changes will be introduced.
> This may necessitate that SDKs which have released `1.0` versions, release `2.0` versions in order to adhere to breaking specification changes.
```